src/Sagas/saga.js

```diff
@@ -53,6 +53,76 @@ export function* watchSelectCity() {
   }
 }
 
+function* watchOnSignUpClick() {
+  while (true) {
+    const action = yield take('ON_SIGNUP_CLICK');
+    yield call(onSignUpClick, action)
+  }
+}
+
+function* onSignUpClick(action) {
+  try {
+     yield call(axios, {
+      method: 'POST',
+      url: `${Constants.HOSTNAME}signup`,
+      data: action.data,
+      config: { headers: {'Content-Type':'application/json'}}
+    });
+    yield put(Actions.signIn({
+      username: action.data.username
+    }));
+  } catch (err) {
+    console.log('sign up unsuccessful')
+  }
+}
+
+function* watchValidateUsername() {
+  while (true) {
+    const action = yield take('VALIDATE_USERNAME_BACKEND');
+    yield call(validateUsername, action)
+  }
+}
+
+function* validateUsername(action) {
+  try {
+    yield call(axios, {
+      method: 'POST',
+      url: `${Constants.HOSTNAME}username`,
+      data: action.data,
+      config: { headers: {'Content-Type':'application/json'}}
+    });
+    yield put(Actions.validateUsernameFrontEnd(true))
```

Comment:
If this function is in this code, `FrontEnd` can be assumed and dropped from the name